lib/Hakyll/Core/Compiler/Internal.hs

```diff
@@ -112,18 +114,19 @@ instance Monoid CompilerWrite where
 
 --------------------------------------------------------------------------------
 -- | Distinguishes reasons in a 'CompilerError'
-data Reason a
-    -- | An exception occured during compilation
-    = CompilationFailure a
-    -- | Absence of any result, most notably in template contexts
-    | NoCompilationResult a
+data CompilerErrors a
+    -- | One or more exceptions occured during compilation
+    = CompilationFailure (NonEmpty a)
+    -- | Absence of any result, most notably in template contexts.  May still
+    -- have error messages.
+    | CompilationNoResult [a]
     deriving Functor
 
 
 -- | Unwrap a `Reason`
-getReason :: Reason a -> a
-getReason (CompilationFailure x)  = x
-getReason (NoCompilationResult x) = x
+compilerErrorMessages :: CompilerErrors a -> [a]
+compilerErrorMessages (CompilationFailure x)  = NonEmpty.toList x
```

Comment:
What's your specific reasoning to avoid empty lists here? And why did `compilerThrow` change to produce a `CompilationNoResult` if no reason value is given?
(Not assuming it's a bad thing, I just don't understand the idea behind it).